shell/browser/api/electron_api_download_item.cc

```diff
@@ -71,65 +97,89 @@ DownloadItem::~DownloadItem() {
     download_item_->RemoveObserver(this);
     download_item_->Remove();
   }
+}
 
-  // Remove from the global map.
-  g_download_item_objects.erase(weak_map_id());
+bool DownloadItem::CheckDestroyed() const {
+  if (!download_item_) {
+    gin_helper::ErrorThrower(v8::Isolate::GetCurrent())
+        .ThrowError("Object has been destroyed");
```

Comment:
Minor suggestion to make life a little easier for people hitting this exception someday:

```suggestion
        .ThrowError("DownloadItem used after being destroyed");
```